compat/getaddrinfo.cc

```diff
@@ -233,8 +227,10 @@ xgetaddrinfo (const char *nodename, const char *servname,
 
     /* For each element pointed to by hp, create an element in the
        result linked list. */
-    sai = eai = nullptr;
-    for (addrs = hp->h_addr_list; *addrs; addrs++) {
+    struct addrinfo *sai = nullptr;
+    struct addrinfo *eai = nullptr;
+    for (char **addrs = hp->h_addr_list; *addrs; addrs++)
+    {
```

Comment:
Please simplify if possible (and avoid unusual/new `{` position):

```suggestion
    for (auto addrs = hp->h_addr_list; *addrs; ++addrs) {
```